components/ClipBoardText.tsx

```diff
@@ -0,0 +1,38 @@
+'use client';
+
+import React, { useState } from 'react';
+import Image from 'next/image';
+import { frame } from '@/public/assets';
+
+interface ClipboardTextProps {
+  text: string;
+}
+
+const ClipboardText: React.FC<ClipboardTextProps> = ({ text }) => {
+  const [isCopied, setIsCopied] = useState<boolean>(false);
+
+  const handleCopyClick = () => {
+    navigator.clipboard
+      .writeText(text)
+      .then(() => {
+        setIsCopied(true);
+        setTimeout(() => setIsCopied(false), 2000);
+      })
+      .catch((err) => {
+        console.error('Failed to copy text: ', err);
```

Comment:
If they only place you find you need to show some alert text for UX, then this is fine. But if you find yourself doing this more then you should integrate toasts